doc/scenario.md

```diff
@@ -0,0 +1,317 @@
+# Role Scenario
+
+The role uses a scenario of steps to determine what to do.
+
+Using a scenario of steps, you can:
+- run the specified steps in any order, but API must be followed (with `cartridge_scenario`);
+- replace the steps of the role with your own or add new steps
+  (with `cartridge_custom_steps_dir` or `cartridge_custom_steps`).
+
+## Steps
+
+In `cartridge_scenario` you should specify names of steps. The default scenario includes the following steps:
+- [deliver_package](#deliver_package)
+- [update_package](#update_package)
+- [update_instance](#update_instance)
+- [configure_instance](#configure_instance)
+- [restart_instance](#restart_instance)
+- [wait_instance_started](#wait_instance_started)
+- [connect_to_membership](#connect_to_membership)
+- [edit_topology](#edit_topology)
+- [cleanup_expelled](#cleanup_expelled)
+- [configure_auth](#configure_auth)
+- [configure_app_config](#configure_app_config)
+- [bootstrap_vshard](#bootstrap_vshard)
+- [configure_failover](#configure_failover)
+
+In addition to the steps in the default scenario, there are additional steps:
+- [set_control_instance](#set_control_instance)
+
+To replace the steps of the role with your own or add new steps, you should use `cartridge_custom_steps_dir`
+or `cartridge_custom_steps` options.
+
+## Example
+
+For example, you can run `edit_topology` step without `connect_to_membership` step (for existing cluster):
+```yaml
+cartridge_scenario:
+  - edit_topology
+  - cleanup_expelled
+```
+
+Also, you can specify control instance in config for `edit_topology` step:
+```yaml
+control_instance:
+  name: core_1
+  console_sock: '/var/run/tarantool/core_1.control'
+```
+
+If you want to replace `deliver_package` step with a custom one, you should set a path to a directory
+with `deliver_package.yml` file to the variable `cartridge_custom_steps_dir`:
+```yaml
+# There is a file "deliver_package.yml" in folder "./custom_steps"
+cartridge_custom_steps_dir: "./custom_steps"
+```
+
```

Comment:
I think that we can show one more example: writing some custom step and adding it to scenario.
Smth like `do_some_very_specific_thing`.

Later, say that if user doesn't want to place all custom steps in one directory (e.g. if he has some steps that are common for a few projects), he can specify a full pach to such steps explicitly using `cartridge_custom_steps`.

Reply:
Yes, I added example for new step and for custom scenario